### PR TITLE
[Sprint 5] Edit List Category Favorite User

### PR DESCRIPTION
**Pull request by Tuanlvt**
Task: Logic Profile
Link redmine: https://dev.framgia.com/redmine/issues/135500

### CODE REVIEW

File: app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt [hunk 1]
```diff
@@ -53,20 +54,17 @@ class CategoryFavoriteFragment : BaseFragment(), CategoryFavoriteContract.ViewMo
...
-    mAdapter.updateData(listCategory)
-  }
-
-  override fun onShowProgressBar() {
-    mIsVisiableProgressBarListCategory.set(true)
-  }
-
-  override fun onDismisProgressBar() {
-    mIsVisiableProgressBarListCategory.set(false)
+  private fun setCurrentView() {
```
Comment (tranminhtri911) [inline app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt]: setCurrentView -> initData

Comment (tranminhtri911) [inline app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt]: setCurrentView -> initData

File: app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt [hunk 2]
```diff
@@ -53,20 +54,17 @@ class CategoryFavoriteFragment : BaseFragment(), CategoryFavoriteContract.ViewMo
...
-  }
-
-  override fun onShowProgressBar() {
-    mIsVisiableProgressBarListCategory.set(true)
-  }
-
-  override fun onDismisProgressBar() {
-    mIsVisiableProgressBarListCategory.set(false)
+  private fun initData() {
+    if (mUserRepository.getUserLocal() != null) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt]: pls change to ``mUserRepository.getUserLocal()?.let {}``

File: app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt [hunk 3]
```diff
@@ -53,20 +54,17 @@ class CategoryFavoriteFragment : BaseFragment(), CategoryFavoriteContract.ViewMo
...
+      mUser.set(mUserRepository.getUserLocal())
+      mAdapter.updateData(mUser.get().categories)
+    }
+    return
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt]: remove return

File: app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt [hunk 4]
```diff
@@ -53,20 +54,17 @@ class CategoryFavoriteFragment : BaseFragment(), CategoryFavoriteContract.ViewMo
...
-    mAdapter.updateData(listCategory)
-  }
-
-  override fun onShowProgressBar() {
-    mIsVisiableProgressBarListCategory.set(true)
-  }
-
-  override fun onDismisProgressBar() {
-    mIsVisiableProgressBarListCategory.set(false)
+  private fun initData() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/categoryfavorite/CategoryFavoriteFragment.kt]: ``initData`` -> ``fillData()``

File: app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt [hunk 5]
```diff
@@ -78,9 +81,12 @@ class MenuProfileFragment : BaseFragment(), MenuProfileContract.ViewModel {
     if (resultCode == Activity.RESULT_OK && requestCode == Constant.RequestCode.TAB_PROFILE_REQUEST) {
       mIsVisibleLayoutNotLoggedIn.set(false)
       mUser.set(mUserRepository.getUserLocal())
-      val fragment = menuProfileAdapter.getFragment(
+      val fragmentPersonalInformation = menuProfileAdapter.getFragment(
           0)?.childFragmentManager?.fragments?.get(0) as PersonalInforFragment
-      fragment.setUser(mUserRepository.getUserLocal())
+      val fragmentCategoryFavorite = menuProfileAdapter.getFragment(
+          1)?.childFragmentManager?.fragments?.get(0) as CategoryFavoriteFragment
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: don't direct cast
Pls check before cast
like
```kt
val fragmentCategoryFavorite = menuProfileAdapter.getFragment(1)?.childFragmentManager?.fragments?.get(0)
if (fragmentCategoryFavorite is CategoryFavoriteFragment) {
fragmentPersonalInformation.updateUser()
}
```

Comment (daolq3012) [reply app/src/main/java/com/framgia/fbook/screen/menuprofile/MenuProfileFragment.kt]: method ``updateUser`` declare getUser from UserRepository

**Pull request merged**